pkg/executor/push.go

```diff
@@ -109,6 +109,12 @@ func DoPush(image v1.Image, opts *config.KanikoOptions) error {
 	t := timing.Start("Total Push Time")
 	var digestByteArray []byte
 	var builder strings.Builder
+
+	parentDir := filepath.Dir(opts.DigestFile)
```

Comment:
Can we move this entire logic into a new function `writeDigestFile(string, []byte)`
```
func writeDigestFile(path string, digestByteArray []byte) error {
    if _, err := os.Stat(filepath.Dir(path)); os.IsNotExist(err) {
	        if err := os.MkdirAll(parentDir, 0700); err != nil {
		  logrus.Debugf("error creating %s, %s", parentDir, err)
                }
                logrus.Tracef("Created directory %v", parentDir)
   }
   return ioutil.WriteFile(opts.DigestFile, digestByteArray, 0644
}
```

Reply:
Sure will update and add unit tests as well :)